src/managers/UserManager.js

```diff
@@ -36,7 +38,8 @@ class UserManager extends BaseManager {
    */
   resolve(user) {
     if (user instanceof GuildMember) return user.user;
-    if (user instanceof Message) return user.author;
+    else if (user instanceof Message) return user.author;
+    else if (typeof user === 'string') user = user.match(USERS_PATTERN)[1];
```

Comment:
I am aware, isn't this not a typechecked arg library? I can of course add an if !pattern.test(user) throw err... but I didn't think that would be necessary?

Reply:
I'm not sure what you mean - the TypeScript types you provided was `string` which would accept any string, including those that don't match the regex (see my other suggestion), so that type-checking wouldn't have worked. You can see no other type-checking is occurring here.

In addition, although it does have typings it is at its core a Javascript library, used by Javascript developers. Type-checking is optional, if developers choose to use TypeScript.

Reply:
That's what I mean, type-checking is optional. Regardless, what would a better way be to handle this then?

Reply:
You're obstructing the functionality that would have passed whatever unchecked input on to `super.resolve(user)` by throwing an error - that's undesirable. Check if the regex returns `null` - if it doesn't, you can return that value. Otherwise allow it to continue.

Reply:
I don't understand, so if it is a valid match, then we should pass the input onto super.resolve()? otherwise we should what?

Reply:
The opposite. *If* its a valid match, return the match. Otherwise let it continue on.

Just stop it from throwing errors.

Reply:
we can't return the match, its still a userid and this function is supposed to return a user object, that would work for the other one, should we just return null here if it's null?

Reply:
Yeah my bad, but yes, only re-assign `user` if you're actually getting a valid match, and then resolve that

Reply:
> Yeah my bad, but yes, only re-assign `user` if you're actually getting a valid match, and then resolve that. Otherwise if mention or Snowflake are the only things left, you might be able to optional chain it.

OHHHHHHH Okay

Reply:
Okay, I think I handle this in the latest commit